convert2rhel/pkghandler.py

```diff
@@ -662,6 +663,30 @@ def remove_non_rhel_kernels():
     return non_rhel_kernels
 
 
+def fix_default_kernel():
+    """
+    Systems converted from Oracle Linux or CentOS Plus may have leftover kernel-uek or kernel-plus in
+    /etc/sysconfig/kernel as DEFAULTKERNEL.
+    This function fixes that by replacing the DEFAULTKERNEL setting from kernel-uek or kernel-plus to rhel-kernel
+    """
+    loggerinst = logging.getLogger(__name__)
+
+    loggerinst.info("Checking for incorrect kernel assignment...")
+    kernel_sys_cfg = utils.get_file_content("/etc/sysconfig/kernel")
+
+    kernel_to_change = ["kernel-uek", "kernel-plus"]
+
+    if any(kernel in kernel_sys_cfg for kernel in kernel_to_change):
+        loggerinst.warning("Incorrect kernel assignment found, fixing...")
+        # need to change to "kernel" in rhel6, 7 and "kernel-core" in rhel8
+        new_kernel_str = "DEFAULTKERNEL=" + ("kernel" if system_info.version.major in [6, 7] else "kernel-core")
+
+        for kernel in kernel_to_change:
+            kernel_sys_cfg = kernel_sys_cfg.replace("DEFAULTKERNEL=" + kernel, new_kernel_str)
```

Comment:
Given my suggestion above:
```suggestion
        kernel_sys_cfg = kernel_sys_cfg.replace("DEFAULTKERNEL=" + kernel_to_change, new_kernel_str)
```